toplevel/coqargs.ml

```diff
@@ -581,4 +582,6 @@ let parse_args arglist : coq_cmdopts * string list =
   in
   try
     parse init_args
-  with any -> fatal_error any
+  with any ->
+    let e = CErrors.push any in
+    fatal_error e
```

Comment:
Why this change?

Reply:
Same reason. I moved `fatal_error` to take an `iexn`, so I needed to turn an `exn` into a `iexn` (this step was previously done in `Topfmt.print_err_exn`.

PS: I'm learning and I'd be happy to know better how to deal with the `iexn` API.